galasa-parent/dev.galasa.framework.api.secrets/src/main/java/dev/galasa/framework/api/secrets/internal/routes/SecretDetailsRoute.java

```diff
@@ -0,0 +1,245 @@
+        QueryParameters queryParams,
+        HttpServletRequest request,
+        HttpServletResponse response
+    ) throws FrameworkException {
+        logger.info("handleGetRequest() entered. Getting secret with the given name");
+        String secretName = getSecretNameFromPath(pathInfo);
+        GalasaSecret secret = getSecretByName(secretName);
+
+        logger.info("handleGetRequest() exiting");
+        return getResponseBuilder().buildResponse(request, response, "application/json",
+            gson.toJson(secret), HttpServletResponse.SC_OK);
+    }
+
+
+    @Override
+    public HttpServletResponse handlePutRequest(
+        String pathInfo,
+        QueryParameters queryParams,
+        HttpServletRequest request,
+        HttpServletResponse response
+    ) throws FrameworkException, IOException {
+        logger.info("handlePutRequest() entered. Validating request payload");
+        checkRequestHasContent(request);
+
+        String secretName = getSecretNameFromPath(pathInfo);
+        SecretRequest secretPayload = parseRequestBody(request, SecretRequest.class);
+
+        ICredentials existingSecret = credentialsService.getCredentials(secretName);
+        GalasaSecretType existingSecretType = getSecretType(existingSecret);
+        validateUpdateRequest(existingSecretType, secretPayload, existingSecret);
+
+        ICredentials decodedSecret = null;
+        int responseCode = HttpServletResponse.SC_NO_CONTENT;
+        if (existingSecret == null) {
+            // No secret with the given name exists, so create a new one
+            decodedSecret = decodeCredentialsFromSecretPayload(secretPayload);
+            responseCode = HttpServletResponse.SC_CREATED;
+        } else if (secretPayload.gettype() != null) {
+            // When a secret type is given, all relevant fields for that type are required,
+            // so overwrite the existing secret to change its type
+            decodedSecret = decodeCredentialsFromSecretPayload(secretPayload);
+        } else {
+            // A secret already exists and no type was given, so just update the secret by
+            // overriding its existing values with the values provided in the request
+            decodedSecret = getOverriddenSecret(existingSecretType, existingSecret, secretPayload);
+        }
+        credentialsService.setCredentials(secretName, decodedSecret);
+
+        logger.info("handlePutRequest() exiting");
+        return getResponseBuilder().buildResponse(request, response, responseCode);
+    }
+
+    @Override
+    public HttpServletResponse handleDeleteRequest(
+        String pathInfo,
+        QueryParameters queryParams,
+        HttpServletRequest request,
+        HttpServletResponse response
+    ) throws FrameworkException {
+        logger.info("handleDeleteRequest() entered");
+        // The provided name is implicitly validated by the route's regex pattern
+        String secretName = getSecretNameFromPath(pathInfo);
```

Comment:
We get the secret name but don't validate it before we pass it onto couchdb ?

Reply:
By the time this code is hit, the secret name has been validated by the route's regex pattern (as noted in the comment) - this currently accepts only alphanumeric characters, underscores, and dots. If the validation fails, the route handling code won't pass anything to etcd.

Happy to add more validation here, but if we'll be performing the same checks as the route's regex, the checks will always pass.